test/e2e/specs/interactivity/router-regions.spec.ts

```diff
@@ -97,4 +97,21 @@ test.describe( 'Router regions', () => {
 		await page.getByTestId( 'back' ).click();
 		await expect( nestedRegionSsr ).toHaveText( 'content from page 1' );
 	} );
+
+	test( 'Page title is updated', async ( { page } ) => {
+		expect( await page.title() ).toBe(
+			'router regions – page 1 – gutenberg'
+		);
+		await page.getByTestId( 'next' ).click();
+		await page.waitForFunction(
+			() => document.title === 'router regions – page 2 – gutenberg'
+		);
+		expect( await page.title() ).toBe(
+			'router regions – page 2 – gutenberg'
+		);
+		await page.getByTestId( 'back' ).click();
+		expect( await page.title() ).toBe(
+			'router regions – page 1 – gutenberg'
+		);
```

Comment:
I think it would be better to use `assert( page ).toHaveTitle()` instead of getting the title and comparing its value directly. That way, we wouldn't need to use `page.waitForFunction()`. 🤔 

```suggestion
		await expect( page ).toHaveTitle(
			'router regions – page 1 – gutenberg'
		);
		await page.getByTestId( 'next' ).click();
		await expect( page ).toHaveTitle(
			'router regions – page 2 – gutenberg'
		);
		await page.getByTestId( 'back' ).click();
		await expect( page ).toHaveTitle(
			'router regions – page 1 – gutenberg'
		);
```